docs/waveform_plugin.rst

```diff
@@ -0,0 +1,91 @@
+.. _waveform_plugin:
+
+------------------------------------------------------
+Making new waveform approximants available to PyCBC
+------------------------------------------------------
+
+=================================================
+Adding a custom waveform model within a script
+=================================================
+
+By example, the following script shows how to write a waveform model
+in the form required for PyCBC. We can also make this new waveform directly
+accessible by using the `pycbc.waveform.plugin.add_custom_waveform` function.
+If you are developing in a notebook or self-contained script, this may be
+what you want to do. However, if you want to make your waveform available
+to pycbc-based executables such as PyCBC Inference, also read the next
+section. 
+
+Ther are two kinds of models you can make. In the example below, we
+make a time-domain model. You can also make a freuqency-domain model. The only
+difference is that your function should return pycbc FrequencySeries and
+the required sample step option is 'delta_f' instead of 'delta_t'. 
+
+Each waveform generation function must take only keyword arguments, and
+should be able to take an arbitrary number of them. You may add new parameters
+as you like. These will be automatically useable by PyCBC Inference and
+other pycbc codes.
+
+Each waveform model must have an associate 'approximant' name, which identifies
+the model and distinguishes it from any other. If the name has already been
+used, you may need to select a different name. 
```

Comment:
Change "may need" to "should"? Otherwise, you should address what happens if you use a name that's already used. (I'm guessing it would override it, since it looks like the custom approximants are being added to the cpu dicts last, but maybe not. Might be best to just to tell people not to do that.) Also, it would be good to point people to the docs [here](http://pycbc.org/pycbc/latest/html/waveform.html#what-waveforms-can-i-generate) so they know how to see what approximant names are currently used.

Reply:
It raises an error by default.